### DISCUSSION TITLE
Can I integrate yazi with lsp?

### DISCUSSION THREAD

**Discussion by Demianeen** [sxyazi/yazi#755] [1. Q&A] [answered]
I use neovim and I use mostly yazi for navigating between files, just because it is soo good)

The only feature that stops me from deleting the nvim-tree though is that yazi does not invoke my nvim lsp client.

I know that yazi is a completely different app from neovim, but is it possible somehow to connect yazi with Neovim's lsp client?

**Comment by sxyazi** [comment]
Could you be more specific please? What tasks do you want to accomplish through LSP?

**Reply by Demianeen** [reply]
Sure! I mean autopaths updates on file/folder rename and on file/folder move. Do you think it is possible?

**Reply by sxyazi** [reply, marked as answer]
I think it's possible that Yazi can output a list of files with user-renamed files, and then the Yazi plugin in Neovim can notify Nvim LSP to update the corresponding files based on that list.

Here is an issue about it: https://github.com/sxyazi/yazi/issues/766. I'll first implement the Yazi side of the work (outputting these files) in the coming days, and then Yazi Neovim plugins will need corresponding adaptations, which will be a complex task. 😄

**Reply by sxyazi** [reply]
Hi, a new [yazi.nvim plugin](https://github.com/mikavilpas/yazi.nvim) written by @mikavilpas now [supports it](https://github.com/mikavilpas/yazi.nvim/commit/43ed7dcd24fc6e9ecbc8b74bc934d5619fce0c12). Feel free to try it out. (Requires the latest Yazi code atm)

**Reply by mikavilpas** [reply]
This is so cool, I hadn't realized someone had suggested this earlier 😆 
I had the exact same idea the day before yesterday and I managed to hack together an implementation for this yesterday.

**Reply by mikavilpas** [reply]
For those that are interested, I drew a fancy diagram that explains how this feature works.

https://github.com/mikavilpas/yazi.nvim/blob/master/dev-documentation/lsp-renaming.md

It may be useful for other editor integrations, but the real reason was that I wanted to draw something fancy 🙂

**Reply by sxyazi** [reply]
Sweet, thank you for the excellent work @mikavilpas! 👍🏻
